### PR TITLE
Adhere to PEP 685 when evaluating markers with extras

### PR DESCRIPTION
**Pull request by hroncok**
PEP: https://peps.python.org/pep-0685/
Original problem: https://discuss.python.org/t/7614

Part of #526

### CODE REVIEW

**Review by pradyunsg — APPROVED** [review]

File: packaging/markers.py [hunk 1]
```diff
@@ -219,6 +220,16 @@ def _get_env(environment: Dict[str, str], name: str) -> str:
     return value
 
 
+def _normalize(*values: str, key: str) -> Tuple[str, ...]:
+    # PEP 685 – Comparison of extra names for optional distribution dependencies
+    # https://peps.python.org/pep-0685/
+    if key == "extra":
+        return tuple(re.sub(r"[-_.]+", "-", v).lower() for v in values)
```
Comment (brettcannon) [inline packaging/markers.py:227]: I don't think we need to re-implement the normalization when `.utils.canonicalize_name()` does the work already.

```suggestion
        return tuple(map(canonicalize_name, values))
```

**Review by brettcannon — CHANGES_REQUESTED** [review]
LGTM, just want to simplify the code a bit.

**Comment by brettcannon** [comment]
FYI I have an in-progress PR that also normalizes the value so that the string representation is fully normalized. But I like the solution for this part of it by @hroncok more than mine. 🙂

File: packaging/markers.py (hunk 1 above)
Comment (hroncok) [reply packaging/markers.py:227]: Awesome. I still prefer a comprehension over map if that's fine with you.

Comment (hroncok) [reply packaging/markers.py]: Added a fixup commit.

**Comment by hroncok** [comment]
There is a fixup commit now. Please squash when (or before) merging or tell me to do it when ready.

**Comment by pradyunsg** [comment]
Please go ahead and squash.

I learnt this recently: If you make a force-push with a single commit amended, it's possible to see what you've changed in the force-push -- by clicking "force-pushed" text in the event that GitHub adds. This is useful for exactly the situation we have here -- a single commit fixup. :)

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
The change LGTM, so like Pradyun said, please go ahead and squash. 🙂

**Comment by hroncok** [comment]
Done.

**Pull request merged**